packages/artifact/src/internal-config-variables.ts

```diff
@@ -10,6 +12,16 @@ export function getUploadChunkSize(): number {
   return 4 * 1024 * 1024 // 4 MB Chunks
 }
 
+export function getDownloadFileConcurrency(): number {
+  // limit the number of concurrent file downloads per artifact to the number of CPU cores available
+  return cpus().length
```

Comment:
We should still do 2 here.  The concern isn't about the client - it's about how many connections we open on the server.

Reply:
For v1 though, the processor count is used to control how many concurrent downloads are going on: https://github.com/actions/runner/blob/6c70d53eead402ba5d53676d6ed649a04e219c9b/src/Runner.Plugins/Artifact/FileContainerServer.cs#L171

Reply:
Maybe that's why we see those high cpu/db load... :D

Reply:
Perhaps 🙄 I'll bump it down to 2, we can always adjust the number in the future